src/Label2.tsx

```diff
@@ -0,0 +1,114 @@
+import React from 'react'
+import styled, {css} from 'styled-components'
+import {get} from './constants'
+
+interface Props {
+  /** The color of the label */
+  variant?: LabelColorOptions
+  /** How large the label is rendered */
+  size?: LabelSizeKeys
+}
+interface LabelColorConfig {
+  borderColor: (props: Props) => React.CSSProperties['color']
+  textColor?: (props: Props) => React.CSSProperties['color']
+}
+export type LabelColorOptions =
+  | 'default'
+  | 'primary'
+  | 'secondary'
+  | 'accent'
+  | 'success'
+  | 'attention'
+  | 'severe'
+  | 'danger'
+  | 'done'
+  | 'sponsors'
+
+type LabelSizeKeys = 'small' | 'large'
+
+export const labelColorMap: Record<LabelColorOptions, LabelColorConfig> = {
+  default: {
+    borderColor: get('colors.border.default')
+  },
```

Comment:
I wonder if we can use the [`variants`](https://styled-system.com/variants) utility from styled-system to clean up all the `get()` calls here.

Reply:
Maybe something like this:

```jsx
import styled from 'styled-components'
import {sx, SxProp} from './sx'
import { variant, SystemStyleObject } from 'styled-system'

const variants = {
  default: {
    borderColor: 'border.default'
  },
  primary: {
    borderColor: 'fg.default'
  },
  ...
} as const

const sizes = {
  small: {
    height: '20px',
    padding: '7px',
  },
  large: ...
} as const

export type LabelProps = {
  variant: keyof typeof variants
  size: keyof typeof sizes
} & SxProp

const Label = styled.span<LabelProps>`
  ...styles...
  ${variant({ variants })};
  ${variant({ prop: 'size', variants: sizes })};
  ${sx};
`

Label.defaultProps = {
  variant: 'default',
  size: 'small',
}

Reply:
Ah - yea, this feels cleaner. I'll give it a shot 👍